podcasts/Onboarding/Plus/UpgradeLandingView.swift

```diff
@@ -0,0 +1,324 @@
+import SwiftUI
+
+struct UpgradeLandingView: View {
+    @EnvironmentObject var viewModel: PlusLandingViewModel
+
+    private let tiers: [UpgradeTier] = [.plus, .patron]
+
+    @State private var currentPage: Int = 0
+
+    @State private var displayPrice: DisplayPrice = .yearly
+
+    var body: some View {
+        VStack(spacing: 0) {
+            topBar
+
+            ZStack {
+
+                if currentPage == 0 {
+                    tiers[0].background
+                        .transition(.opacity.animation(currentPage == 0 ? .easeIn : .easeOut))
+                        .ignoresSafeArea()
+                } else {
+                    tiers[1].background
+                        .transition(.opacity.animation(currentPage == 0 ? .easeIn : .easeOut))
+                        .ignoresSafeArea()
+                }
+
+                VStack(spacing: 0) {
+                    ScrollViewIfNeeded {
+                        VStack(spacing: 0) {
+
+
+                            PlusLabel(currentPage == 0 ? L10n.plusMarketingTitle : L10n.patronCallout, for: .title2)
+                                .minimumScaleFactor(0.5)
+                                .lineLimit(2)
+                                .padding(.bottom, 16)
+                                .padding(.horizontal, 32)
+
+                            UpgradeRoundedSegmentedControl(selected: $displayPrice)
+                                .padding(.bottom, 24)
+
+                            FeaturesCarousel(currentIndex: $currentPage.animation(), currentPrice: $displayPrice, tiers: tiers)
+
+                            PageIndicator(numberOfItems: tiers.count, currentPage: currentPage)
+                            .padding(.top, 27)
+                        }
+                    }
+                }
+            }
+        }
+        .background(Color(hex: "121212"))
+    }
+
+    var topBar: some View {
+        HStack(spacing: 0) {
+            Rectangle()
+                .foregroundColor(.clear)
+                .frame(height: 44)
+            Button(viewModel.source == .upsell ? L10n.eoyNotNow : L10n.plusSkip) {
+                viewModel.dismissTapped()
+            }
+            .foregroundColor(.white)
+            .font(style: .body, weight: .medium)
+            .padding(.trailing)
+        }
+    }
+
+    enum DisplayPrice {
+        case yearly, monthly
+    }
+}
+
+// MARK: - Feature Carousel
+
+struct FeaturesCarousel: View {
```

Comment:
wdyt about making all the structs that are used only on this view private to prevent them from showing up in the globally?

```suggestion
private struct FeaturesCarousel: View {
```

Reply:
Done in b1edb48243e9242810904e305c783bff323c42e9